dm/worker/subtask_holder.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"sync"
+)
+
+// subTaskHolder holds subtask instances.
+type subTaskHolder struct {
+	// protect concurrent access of `subTasks`
+	// the internal operation of the subtask should be protected by any mechanism in the subtask itself.
+	mu       sync.RWMutex
```

Comment:
I personally tend to use this style, because of the methods from `sync.RWMutex` will not be exported to the outer space.

In this case, `Worker` -> `subTaskHolder` -> `mu`, then `Worker` can not call `mu`'s method, like `w.subTaskHolder.Lock`.

BTW, seems the builtin pkgs prefer this style? maybe we can define our style later?